### PR TITLE
a user can create a tag

### PR DESCRIPTION
**Pull request by SharifRashed**
#### Changes Made
1. Added file `tagForm.js` to `Tag` directory.
2. Added to `ApplicationViews.js` file to include proper  routing for new tag.
3. Refactored `TagList.js` file to display `a list of tags` instead of using `tag.js`.
4. Refactored `TagController.cs`, `Tag.cs` and `TagRepository.cs` to be more uniform
​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout sr-create-tag
    ```
3. Open a new Terminal tab (⌘T) and navigate to the server directory.
5. Test app functionality.
    > Inside your `Client directory`
            `Npm Start`
    > Run the `web api`
    > Example: When user does `submits a new tag`, then `the user should be directed to a list of tags`.
6. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

### CODE REVIEW

**Review by JacobF94 — APPROVED** [review]
Adding tags functional

File: Tabloid/client/src/components/ApplicationViews.js [hunk 1]
```diff
@@ -26,7 +27,7 @@ export default function ApplicationViews({ isLoggedIn }) {
           <CategoryForm />
         </Route>
         <Route path="/categories/delete/:id" exact>
```
Comment (heymonicakay) [inline Tabloid/client/src/components/ApplicationViews.js:40]: Does this route need to be exact?

Comment (heymonicakay) [reply Tabloid/client/src/components/ApplicationViews.js:40]: Should the id param in this route be an integer? if so, please make sure it's an integer by adding regex to the param. See my suggestion ...

Comment (heymonicakay) [reply Tabloid/client/src/components/ApplicationViews.js:40]:
```suggestion
        <Route path="/categories/delete/:id(\d+)">
```

File: Tabloid/client/src/components/Tag/tagForm.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        });
+    };
+
+    return (
+        <Form>
+            <FormGroup>
+                <Label for="name">Name</Label>
+                <Input type="text" name="name" id="name" placeholder="tag name" value={tag.name} onChange={handleInputChange} />
+            </FormGroup>
+            <Button className="btn btn-primary" onClick={handleSave}>Submit</Button>
```
Comment (heymonicakay) [inline Tabloid/client/src/components/Tag/tagForm.js]: Consider adding some kind of form validation here to disable the button until the user has filled out the form input completely.

File: Tabloid/client/src/components/Tag/tagForm.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import { FormGroup, Input, Label, Button, Form } from "reactstrap";
+import { addTag } from "../../modules/tagManager";
+
+const TagForm = () => {
+    const history = useHistory();
+    const emptyTag = {
+        Name: "",
```
Comment (heymonicakay) [inline Tabloid/client/src/components/Tag/tagForm.js]: Watch your casing here. Client side code (especially in React) is camel cased.

File: Tabloid/client/src/components/Tag/tagForm.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        const key = evt.target.id;
+
+        const tagCopy = { ...tag };
+
+        tagCopy[key] = value;
+        setTag(tagCopy);
+    };
+    const handleSave = (evt) => {
+        evt.preventDefault();
+        addTag(tag).then((p) => {
```
Comment (heymonicakay) [inline Tabloid/client/src/components/Tag/tagForm.js]: do you need a parameter (p) in this callback function?

File: Tabloid/client/src/components/ApplicationViews.js [hunk 5]
```diff
@@ -41,6 +42,10 @@ export default function ApplicationViews({ isLoggedIn }) {
           <TagList />
         </Route>
 
+        <Route path="/tags/new" exact>
```
Comment (heymonicakay) [inline Tabloid/client/src/components/ApplicationViews.js]: does this route need to be exact?

**Review by heymonicakay — CHANGES_REQUESTED** [review]
Please see my comments!

File: Tabloid/client/src/components/ApplicationViews.js (hunk 1 above)
Comment (SharifRashed) [reply Tabloid/client/src/components/ApplicationViews.js:40]: 👍🏻

File: Tabloid/client/src/components/Tag/tagForm.js (hunk 4 above)
Comment (SharifRashed) [reply Tabloid/client/src/components/Tag/tagForm.js]: 👍🏻

File: Tabloid/client/src/components/Tag/tagForm.js (hunk 2 above)
Comment (SharifRashed) [reply Tabloid/client/src/components/Tag/tagForm.js]: 👍🏻

File: Tabloid/client/src/components/Tag/tagForm.js (hunk 3 above)
Comment (SharifRashed) [reply Tabloid/client/src/components/Tag/tagForm.js]: 👍🏻

File: Tabloid/client/src/components/ApplicationViews.js (hunk 5 above)
Comment (SharifRashed) [reply Tabloid/client/src/components/ApplicationViews.js]: 👍🏻

**Review by heymonicakay — APPROVED** [review]

**Pull request merged**